src/algebra/finite_map/mod.rs

```diff
@@ -39,16 +48,32 @@ where
     /// Find the value associated to the specified key
     fn lookup(&self, key: &Key) -> Value;
 
-    /// Return the set of values that are mapped to non-zero values.
-    fn support(&self) -> SupportIter<'_, Key, Value>;
+    /// Find the value associated to the specified key
+    ///
+    /// Return value `None` indicates that the key is not stored in the map,
+    /// and its associated value is `Value::zero()`.
+    fn get(&self, key: &Key) -> Option<&Value>;
```

Comment:
but this is wrong. FiniteMaps always map to a group, and if a value is not there it is effectively mapped to 0.

Reply:
FiniteMaps are total functions, so they always return a value.

Reply:
you can have a private helper function, but this should not be part of the trait.

Reply:
Renaming to `get_in_support`.